### ISSUE TITLE
KfRaiseIrql usersim API doesn't accurately match kernel mode 

### ISSUE THREAD

**Issue by Alan-Jowett** [microsoft/usersim#184]
In kernel mode, this is a relatively light weight API. The usersim version is very heavy and causes tests to timeout due to cost.

**Comment by Alan-Jowett** [comment]
![image](https://github.com/microsoft/usersim/assets/20480683/93c9aeb9-da98-4ef4-b0ca-25c7261e953d)

**Comment by mtfriesen** [comment]
Nit with title: the performance might not match, but the performance is poor partly because it tries very hard to match the observable functional behavior of kernel mode. Do you know what the bottleneck is?

**Comment by Alan-Jowett** [comment]
Looks like SetThreadPriority and SetThreadGroupAffinity are super slow. I will try profiling to narrow down why.

**Comment by Alan-Jowett** [comment]
Correct. The test case that breaks is running BPF programs at passive level, which requires raising and lowering IRQL on every epoch enter / exit.

**Comment by mtfriesen** [comment]
For context, I added a considerable amount of overhead in PR https://github.com/microsoft/usersim/pull/53. Previously this routine was somewhat lighter weight, but also allowed a variety of observable misbehavior, including passive threads and other DPCs preempting dispatch level code on the same processor.

Perhaps we could relax some of those constraints if the tests do not depend on dispatch level code being properly serialized WRT each other and/or WRT passive threads.
